### DISCUSSION TITLE
[V4] How to add ```@supports``` queries in tailwindcss @theme config

### DISCUSSION THREAD

**Discussion by Stephan-MC** [tailwindlabs/tailwindcss#15398] [Help] [answered]
in [v3] or using tailwind.config.{ts,js}, it can be done as shown below
```ts
import type { Config } from 'tailwindcss';

export default {
  content: ['./projects/**/*.{ts,html}'],
  theme: {
    extend: {
      supports: {
        dvh: 'height: 100dvh',
        dvw: 'width: 100dvw',
      },
    },
  },
} satisfies Config;

```

**Comment by Stephan-MC** [comment]
> in [v3] or using tailwind.config.{ts,js}, it can be done as shown below
> 
> ```ts
> import type { Config } from 'tailwindcss';
> 
> export default {
>   content: ['./projects/**/*.{ts,html}'],
>   theme: {
>     extend: {
>       supports: {
>         dvh: 'height: 100dvh',
>         dvw: 'width: 100dvw',
>       },
>     },
>   },
> } satisfies Config;
> ```

I'm trying to escape from js/ts config files as much as possible. will like to know if there's a way to work around this.

**Reply by Stephan-MC** [reply, marked as answer]
I think i found a solution my self. from the [documentation](https://tailwindcss.com/docs/v4-beta#adding-custom-variants)

```scss
@variant support-dvh (@supports(height: 100dvh))
@variant support-dvw (@supports(width: 100dvw))
```

**Reply by adamwathan** [reply]
Yep that’s the way to do it!
